libc/src/__support/math_extras.h

```diff
@@ -146,8 +146,7 @@ first_trailing_zero(T value) {
 template <typename T>
 [[nodiscard]] LIBC_INLINE constexpr cpp::enable_if_t<cpp::is_unsigned_v<T>, int>
 first_trailing_one(T value) {
-  return value == cpp::numeric_limits<T>::max() ? 0
-                                                : cpp::countr_zero(value) + 1;
+  return value == T(0) ? 0 : cpp::countr_zero(value) + 1;
```

Comment:
Because I wanted to avoid the integral promotion in small integer types, which creates an edge case in systems where `USHRT_WIDTH == INT_WIDTH`. Do you want me to change this one to `first_trailing_zero` or the `big_int` version to the ternary expression? Also, thanks for the review.

Reply:
how about `first_trailing_zero(static_cast<T>(~value))` ?

Reply:
So I should just make the change @lntue suggested in `math_extras.h` and if someone wants to port LLVM-libc to some kind of platform, in which `short` has the same width as `int` (so, we'll just have to try to be aware in the future if there is a port to such architecture, it's usually 16-bit MCUs or devices where `CHAR_BIT` is 16, 24 or 32), they can just change it then? Am I getting this right?

Reply:
I'm not concerned with platforms where `sizeof(short) == sizeof(int)`; if we need to support those, ever, there will need to be other changes throughout llvm-libc. At that point, I'd prefer to know what platforms those are, and who cares about them, such that we can then stand up proper CI upstream for ensuring we don't break them.